### DISCUSSION TITLE
I made a Docker image to develop and run Gleam apps

### DISCUSSION THREAD

**Discussion by Billuc** [gleam-lang/gleam#2706] [Show and tell]
Hello there,

I started to get interested in Gleam since the v1 release and I wanted to experiment with it.
I am a huge fan of Docker and I usually work in Docker environments.
I was surprised to see that there wasn't any Gleam Docker image (more recent that 2 years old), so I did one myself.
[Here](https://hub.docker.com/repository/docker/billuc582/gleam/general) it is ! Feel free to use it.

Here is the Dockerfile : 

```
FROM alpine:latest as base

WORKDIR /tmp

# Install erlang
RUN apk add erlang erlang-dev
RUN apk add wget

# Install rebar
RUN wget https://s3.amazonaws.com/rebar3/rebar3
RUN chmod u+x ./rebar3
RUN ./rebar3 local install
RUN rm ./rebar3

# Install gleam
RUN wget https://github.com/gleam-lang/gleam/releases/download/v1.0.0/gleam-v1.0.0-aarch64-unknown-linux-musl.tar.gz
RUN tar xzf gleam-v1.0.0-aarch64-unknown-linux-musl.tar.gz
RUN mkdir /root/.cache/gleam
RUN mkdir /root/.cache/gleam/bin
RUN mv gleam /root/.cache/gleam/bin
RUN chmod u+x /root/.cache/gleam/bin/gleam
RUN rm gleam-v1.0.0-aarch64-unknown-linux-musl.tar.gz

ENV PATH=/root/.cache/gleam/bin:/root/.cache/rebar3/bin:${PATH}

WORKDIR /
```

If you have any idea for improvments or else, feel free ;)

**Comment by lpil** [comment]
Nice! We've got some too: 

https://github.com/gleam-lang/gleam/pkgs/container/gleam

https://github.com/gleam-lang/gleam/tree/main/containers

**Reply by Billuc** [reply]
Oh yeah, I didn't think about Github's Container Repository ^^'
I always go to the Docker Hub when I am looking for image.

Thanks for pointing me to this :)

**Reply by lpil** [reply]
Yeah we had to stop using them because they started extorting money from open source organisations, threatening to delete all your images if you don't pay.

I'd recommend not supporting the docker company, they're very hostile towards open source. All their competitors are much better.

**Reply by Billuc** [reply]
Oh wow, I didn't know that ! Thanks for sharing !
